storage/ndb/src/kernel/blocks/dblqh/DblqhMain.cpp

```diff
@@ -19704,7 +19805,33 @@ void Dblqh::scanTupkeyConfLab(Signal* signal,
     // Inform API about keyinfo len as well
     read_len += sendKeyinfo20(signal, scanPtr, regTcPtr);
   }//if
-  ndbrequire(scanPtr->m_curr_batch_size_rows < MAX_PARALLEL_OP_PER_SCAN);
+
+  if (scanPtr->m_aggregation) {
+    if (conf->agg_batch_size_bytes) {
```

Comment:
This entire ndbrequire statement + if-statement can be turned into a large
ndbrequire-statement.